### PR TITLE
[TIMOB-24446](6_1_X) Allow replacing bundled Android Support Libraries

### PR DESCRIPTION
**Pull request by janvennemann**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-24446

**Optional Description:**
6_1_X backport of #9008

### CODE REVIEW

File: android/cli/commands/_build.js [hunk 1]
```diff
@@ -3065,7 +3073,13 @@ AndroidBuilder.prototype.copyModuleResources = function copyModuleResources(next
 					resPkgFile = jarFile.replace(/\.jar$/, '.respackage');
 
 				if (fs.existsSync(resPkgFile) && fs.existsSync(resFile)) {
-					this.resPackages[resFile] = fs.readFileSync(resPkgFile).toString().split('\n').shift().trim();
+					var packageName = fs.readFileSync(resPkgFile).toString().split('\n').shift().trim();
```
Comment (garymathews) [inline android/cli/commands/_build.js]: Maybe support both line-endings? `split(/\r?\n/)`

File: android/cli/commands/_buildModule.js [hunk 2]
```diff
@@ -422,8 +443,13 @@ AndroidModuleBuilder.prototype.processResources = function processResources(next
 						var resArchivePathAndFilename = path.join(modulesPath, file.replace(/\.jar$/, '.res.zip'));
 						var respackagePathAndFilename = path.join(modulesPath, file.replace(/\.jar$/, '.respackage'));
 						if (fs.existsSync(resArchivePathAndFilename) && fs.existsSync(respackagePathAndFilename)) {
-							extraPackages.push(fs.readFileSync(respackagePathAndFilename).toString().split('\n').shift().trim());
-							resArchives.push(resArchivePathAndFilename);
+							var packageName = fs.readFileSync(respackagePathAndFilename).toString().split('\n').shift().trim();
```
Comment (garymathews) [inline android/cli/commands/_buildModule.js]: Same here

**Review by garymathews — APPROVED** [review]
CR: **PASS** (support both line endings)

👍

**Comment by lokeshchdhry** [comment]
FR Passed.

Used the steps in the ticket to do the FR.

Studio Ver:     4.9.0.201705180402
SDK Ver:        6.1.0.v20170519131839
OS Ver:         10.12.3
Xcode Ver:      Xcode 8.3.2
Appc NPM:       4.2.9
Appc CLI:       6.2.1
Ti CLI Ver:     5.0.13
Alloy Ver:      1.9.11
Node Ver:       6.10.1
Java Ver:       1.8.0_101
Devices:        ⇨ oneplus A0001 --- Android 6.0.1

**Pull request merged**